### PR TITLE
chore: fine-tune codeowners

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** N/A

## Summary

Updates codeowners file to use reviewer group as catch-all.

### Questions

Do we want to also set up granular ownership? Something like this would work:

```
# component-specific owners
calcite-block/                    @jcfranco
calcite-block-section/            @jcfranco
calcite-color-picker/             @jcfranco
calcite-color-picker-hex-input/   @jcfranco
calcite-color-picker-swatch/      @jcfranco
calcite-icon/                     @jcfranco
calcite-option/                   @jcfranco
calcite-option-group/             @jcfranco
calcite-pick-list/                @jcfranco
calcite-select/                   @jcfranco
calcite-value-list/               @jcfranco
```

Anything not specified above would be owned by the reviewers group (see [codeowners syntax](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax)).

### CODE REVIEW

**Comment by driskull** [comment]
> Do we want to also set up granular ownership? Something like this would work:

Maybe that might be a little too fine grained? I'm ok with it though.

File: CODEOWNERS
Comment (driskull) [inline CODEOWNERS:2]: @julio8a @jcfranco  can we rename this group to `calcite-codeowners`?

**Review by driskull — APPROVED** [review]
This is a good start to slim down notification 👍

**Comment by paulcpederson** [comment]
> Do we want to also set up granular ownership?

I think this would be great in the future. This is at least an improvement, though.

File: CODEOWNERS
Comment (jcfranco) [reply CODEOWNERS:2]: I'm fine either way. Something that applies to reviews and code ownership works for me.

**Comment by jcfranco** [comment]
Merging to slim down notifications. Will hold off on granular ownership for now.

@julio8a If you rename the reviewers group, can you update this file as well? 🙇‍♂️

**Comment by julio8a** [comment]
okay, this looks good to me, but it's good to know this needs to change too :D

**Pull request merged**
